Use non-localized date parsing to fix #1838

ChristofKaufmann opened the pull request:
This commit uses non-localized date parsing in [`fontTools.misc.timeTools.timestampFromString`](https://github.com/fonttools/fonttools/blob/51bff6cf6a08976c831aa5c9e8c901fac7e61dea/Lib/fontTools/misc/timeTools.py#L46). This is consistent with the opposite function `fontTools.misc.timeTools.timestampToString`, which also uses a non-localized date format.

This function is used when importing a font from XML using [`fontTools.ttLib.TTFont.importXML`](https://github.com/fonttools/fonttools/blob/51bff6cf6a08976c831aa5c9e8c901fac7e61dea/Lib/fontTools/ttLib/ttFont.py#L318) via [`fontTools.ttLib.tables.table__h_e_a_d.fromXML`](https://github.com/fonttools/fonttools/blob/51bff6cf6a08976c831aa5c9e8c901fac7e61dea/Lib/fontTools/ttLib/tables/_h_e_a_d.py#L107)

Depending on the LC_TIME locale setting, the parsing of the created and modified dates of a font might fail, as described in #1838. This fix provides simple code to parse the date string with a fixed format, defined by [`fontTools.misc.timeTools.asctime`](https://github.com/fonttools/fonttools/blob/51bff6cf6a08976c831aa5c9e8c901fac7e61dea/Lib/fontTools/misc/timeTools.py#L17). The parsing is now independent of the LC_TIME locale setting.

anthrotype commented:
@ChristofKaufmann thanks for this!
Would you be able to add a test by temporarily doing `locale.setlocale` and verifying that it works?

ChristofKaufmann commented:
I tested this on my machine, but setting the locale requires the locale to be installed on the system. So it's not so easy to test in general.

I also ran `run-tests.sh` and got:
```
========= 3407 passed, 6 skipped, 2 xfailed, 20 warnings in 32.35s ========
_________________________________ summary _________________________________
  py: commands succeeded
  congratulations :)
```

anthrotype commented:
have you tried to see if Travis CI comes with a non-english locale to test with?
If we can't test this easily, we may content ourselves of your manual checks.

ChristofKaufmann commented:
Added `language-pack-de` in Travis CI and added a test function. The test will be skipped, when the locale is not available in the system.

I also tested it with the fixed version of `timestampFromString` and with the broken version. The test behaves as expected. Seems to be good.

Does the added `language-pack-de` package influences tests on osx? Did Travis CI check on osx already?

anthrotype commented:
No, only affects the Linux builds. I think this is good, we can merge
